pkg/util/chunk/pool.go

```diff
@@ -18,8 +18,44 @@ import (
 	"sync"
 
 	"github.com/pingcap/tidb/pkg/types"
+	"github.com/pingcap/tidb/pkg/util/syncutil"
 )
 
+var (
+	globalChunkPoolMutex syncutil.RWMutex
+	// globalChunkPool is a chunk pool, the key is the init capacity.
+	globalChunkPool = make(map[int]*Pool)
+)
+
+func getChunkFromPool(initCap int, fields []*types.FieldType) *Chunk {
+	globalChunkPoolMutex.RLock()
+	pool, ok := globalChunkPool[initCap]
+	if ok {
+		globalChunkPoolMutex.RUnlock()
+		return pool.GetChunk(fields)
+	}
+	globalChunkPoolMutex.RUnlock()
+	globalChunkPoolMutex.Lock()
+	defer globalChunkPoolMutex.Unlock()
+	globalChunkPool[initCap] = NewPool(initCap)
+	return globalChunkPool[initCap].GetChunk(fields)
```

Comment:
is that we need exactly a matched capacity-size pool, a little bit less-than is adequate?

Reply:
this initCap is the size of bucket in the histogram. I think it can meet the requirement.

Reply:
I add some comments for it.